0000-template.md

```diff
@@ -10,18 +10,48 @@
 
 ## Motivation
 
-> Why are we doing this? What use cases does it support? What is the expected
-outcome?
+> Why are we doing this? What use cases does it support?
+
+> Please provide specific examples. If you say "this would be more flexible" then
+> give an example of something that becomes easier. If you say "this would be make
+> it easier to do X" then give an example of what that looks like today and what's
+> hard about it.
+
+> Is there some concrete issue you cannot accomplish without this?
+> What does it look like to accomplish some set of goals today and how could
+> that be improved?
+> Are there any workarounds that are necessary today?
+> Are there open issues on Github where people would be helped by this?
+
+> Please focus on explaining the motivation so that if this RFC is not accepted,
+> the motivation could be used to develop alternative solutions. In other words,
+> enumerate the constraints you are trying to solve without coupling them too
+> closely to the solution you have in mind.
 
 ## Detailed design
 
-> This is the bulk of the RFC.
+### Technical Background
+
+> There are a lot of ways Svelte is used. It's hosted on different platforms;
+> integrated with different libraries; built with different bundlers, etc. No one
+> person knows everything about all the ways Svelte is used. What does someone who
+> knows about Svelte but hasn't necessarily used anything outside of it need to
+> know? Are there docs you can share?
+
+> How do different libraries or frameworks implement this feature? We can take
+> design inspiration from others who have done this well and improve upon the
+> designs or make them better fit Svelte.
+
+### Implementation
+
+> Explain the design in enough detail for somebody familiar with the framework to
+understand, and for somebody familiar with the implementation to implement. This
+should get into specifics and corner-cases, and include examples of how the feature
+is used. Any new terminology should be defined here.
```

Comment:
You said you wanted people to not focus on solutions too soon, but this sounds like the opposite to me ("this should get into specifics and corner-cases").

Reply:
yeah, this was something I had trouble with. I tried to encourage a lot of detail before getting to this point (e.g. motivation and background). I think at some point the corner cases have to get answered before final approval though. I just tried to update this section a bit more. Take a look and let me know what you think

Reply:
Sounds better! Maybe a final sentence like "you don't need to specify this all up front as many of the implementation details may arise during discussion, but at least put some initial thought into it".

Reply:
I think it would be useful for people to include as much of this as they can, but I don't want it to be a barrier. I updated the wording in this section to make these sound a bit more like suggestions instead of being mandatory. I also updated the PR description to better describe what I was trying to accomplish here